### ISSUE TITLE
Valid words that aren't in aspell/mainstream dictionaries

### ISSUE THREAD

**Issue by peternewman** [codespell-project/codespell#1711]
Piecewise isn't in aspell or cambridge dictionary, however it is used in mathematics: https://en.wikipedia.org/wiki/Piecewise

_Originally posted by @sebweb3r in https://github.com/codespell-project/codespell/pull/1697#issuecomment-702891154_

As we get stricter with the aspell tests, we need somewhere to put words which aren't in aspell and/or mainstream dictionaries, so they won't be checked by the tests, but can be given more human oversight.

I guess a number of these are likely to be technical in nature, but perhaps not exclusively?

Suggestions for names welcome...

**Comment by sebweb3r** [comment]
I see one main problem. What about
```
Mistake->aspell, not in aspell,
```

**Comment by peternewman** [comment]
> I see one main problem. What about
> 
> ```
> Mistake->aspell, not in aspell,
> ```
> 
> We don't want duplicates, but this would put again words in an "unchecked" dictionary.

Agreed @sebweb3r . I think there are two options there, either split the corrections in half, hard to manage and needs extra code. Or leave it as proposed, with neither of them checked. My hope is there would be less than say 100 lines in this file, so the whole lot could be checked once a year if someone was really keen. But either way each new entry could be explicitly checked in any PRs to look for mistakes. Hell, we could even report info logs of the state of these corrections when we run the tests (but not fail them), so you could be targeted at which words actually need checking.

**Comment by sebweb3r** [comment]
> Hell, we could even report info logs of the state of these corrections when we run the tests (but not fail them), so you could be targeted at which words actually need checking.

Yes, we could, but then I would stop #1690 and write the tests differently.
Every dictionary has to have a set of aspell dictionaries given and the options for in_aspell
can be True (all words have to be in aspell), False(the words are not allowed to be in aspell) and None (its not mandatory, that the words are in aspell, but we compile a list of words, that are not in aspell)

F.e. dictionary, False, None, english_dicts, english_dicts 
checks if errors are not in english aspell dicts. For the corrections it doesn't assert, but compiles a list of corrections, that are not in aspell.

**Comment by larsoner** [comment]
#1690 is in, so what do we need now?

Can we just keep it simple and if we are talking about a few likely exceptions, we just add a little variable denylist:
```
ignore_in_aspell = {'whatever', 'we', 'need'}
```
and always ignore these. For example this is how I would fix #1627. If this list grows too large then we can just set the `in_aspell` pairing to False, or make an allowlist `check_in_aspell` set. But I would try the simple solution above first and worry about reengineering things later when needed.

**Comment by vikivivi** [comment]
> we need somewhere to put words which aren't in aspell and/or mainstream dictionaries,

@peternewman A personal aspell dictionary in codespell to overwrite those false positive check by adding those words to aspell personal dictionary text file? Or using ispell with "ibritish-insane" / "iamerican-insane" dictionary + personal dictionary to reduce false positive check ?

Example: here is my first 3 words in my aspell personal dictionary text file.
```text
personal_ws-1.1 en 416
accessor
adaptively
adaptor
```
